wait/host.go

```diff
@@ -584,51 +585,77 @@ func (a *HostAwaitility) WaitForTemplateUpdateRequests(namespace string, count i
-	var notification *toolchainv1alpha1.Notification
+// WaitForNotifications waits until there is an expected number of Notifications available for the provided user and with the notification type and which match the conditions (if provided).
+func (a *HostAwaitility) WaitForNotifications(username, notificationType string, numberOfNotifications int, criteria ...NotificationWaitCriterion) ([]*toolchainv1alpha1.Notification, error) {
+	var notifications []*toolchainv1alpha1.Notification
 	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
-		obj := &toolchainv1alpha1.Notification{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj); err != nil {
-			if errors.IsNotFound(err) {
-				a.T.Logf("waiting for availability of notification '%s'", name)
-				return false, nil
-			}
+		labels := map[string]string{toolchainv1alpha1.NotificationUserNameLabelKey: username, toolchainv1alpha1.NotificationTypeLabelKey: notificationType}
+		opts := client.MatchingLabels(labels)
+		notificationList := &toolchainv1alpha1.NotificationList{}
+		if err :=  a.Client.List(context.TODO(), notificationList, opts);  err != nil {
 			return false, err
 		}
-		for _, match := range criteria {
-			if !match(a, obj) {
-				return false, nil
+
+		actualNotificationCount := len(notificationList.Items)
+		if numberOfNotifications < actualNotificationCount{
+			// There are more notifications than the expected amount
+			a.T.Logf("expected '%d' notifications, but found '%d' notifications", numberOfNotifications, actualNotificationCount)
+			return false, errs.New(fmt.Sprintf("expected %d notifications, but found %d notifications", numberOfNotifications, actualNotificationCount))
+		}
+
+		// If the numberOfNotifications is greater than the current notification count
+		// wait for availability of notification(s).
+		if numberOfNotifications != actualNotificationCount {
+			a.T.Logf("waiting for availability of notification '%s'", username)
+			return false, nil
+		}
+
+		for _, n := range notificationList.Items {
+			for _, match := range criteria {
+				if match(a, n) {
+					a.T.Logf("found notification '%s'", username)
+					notifications = append(notifications, &n)
+				}
 			}
 		}
-		a.T.Logf("found notification '%s'", name)
-		notification = obj
+
+		// The criteria hasn't matched yet
+		if len(notifications) <= 0 {
+			return false, nil
+		}
+
 		return true, nil
 	})
-	return notification, err
+	return notifications, err
 }
 
-// WaitUntilNotificationDeleted waits until the Notification with the given name is deleted (ie, not found)
-func (a *HostAwaitility) WaitUntilNotificationDeleted(name string) error {
+// WaitUntilNotificationDeleted waits until the Notification for the given user is deleted (ie, not found)
+func (a *HostAwaitility) WaitUntilNotificationDeleted(username, notificationType string) error {
```

Comment:
you could rename this func to `WaitUntilNotificationsDeleted` (plural on `Notification)

Reply:
Fixed here: 9836e96a3fe2039ac20a77112f55042428ea443f